File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/1386-Shift2dGrid/1386-Shift2dGrid.java
// Last updated: 8/4/2026, 12:45:19 PM
class Solution {
    public List<List<Integer>> shiftGrid(int[][] grid, int k) {
        List<List<Integer>>result = new ArrayList<>();
        int n = grid.length;
        int m = grid[0].length;

        int[][] ans = new int[n][m];  

        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                int newJ = (j + k) % m;                
                int newI = (i + (j + k) / m) % n;      
                ans[newI][newJ] = grid[i][j];
            }
        }

       for(int[] row : ans){
            List<Integer> temp = new ArrayList<>();
            for(int val : row) temp.add(val);
            result.add(temp);
        }

        return result;

    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/1386-Shift2dGrid/1386-Shift2dGrid.cpp
// Last updated: 8/4/2026, 12:45:12 PM
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int n = grid.size();
        int m = grid[0].size();
        vector<vector<int>>ans(n , vector<int>(m , 0));
        vector<vector<int>>res(n , vector<int>(m , 0));

        for(int i = 0 ; i < n ; i++){
            for(int j = 0 ; j < m ; j++){
                int newj = (j + k)%m;
                int newi = (i+(j+k)/m)%n;
                ans[newi][newj] = grid[i][j];
            }
        }

        for(auto it : ans){
            vector<int>temp;
            for(auto val : it){
                temp.push_back(val);
            }
            res.push_back(temp);
        }

        return ans;
    }
};